backend/dataall/api/Objects/Environment/resolvers.py

```diff
@@ -52,7 +75,13 @@ def create_environment(context: Context, source, input=None):
         )
 
     with context.engine.scoped_session() as session:
-        cdk_role_name = check_environment(context, source, input=input)
+        pivot_role_as_part_of_environment = get_pivot_role_as_part_of_environment(context, source)
```

Comment:
It seems you can move `get_pivot_role_as_part_of_environment` check inside `check_environment` to  have the function that performs all necessary checks and returns the correct role.

Reply:
Done! thank you